dev/tests/scheduler.rs

```diff
@@ -449,6 +452,19 @@ async fn schedule_request_lock() {
 
     let ScheduleResponse::Scheduled { drone: drone2, backend_id: backend2, .. } = r2 else {panic!()};
 
+    //scheduling multiple of the same lock should return the same drone
+    let (
+		Ok(ScheduleResponse::Scheduled { spawned: a, .. }),
+		Ok(ScheduleResponse::Scheduled { spawned: b, .. })
+	) = tokio::join!(
+			mock_agent.schedule_drone(&drone_id, false, Some("speedlock".to_string()))
+		,
+		async {
+			tokio::time::sleep(Duration::from_millis(300)).await;
+			mock_agent.schedule_locked(false, Some("speedlock".to_string())).await
+		}
+	) else { panic!("hello") };
```

Comment:
This will be hard to debug if it catches; instead prefer:
- assign each side of the pair to its own variable
- use a `let _ else {}` for each of them, with a panic that at least tells what the problem is